src/Microsoft.AspNet.StaticFiles/HtmlDirectoryFormatter.cs

```diff
@@ -34,6 +37,11 @@ public virtual Task GenerateContentAsync(HttpContext context, IEnumerable<IFileI
                 throw new ArgumentNullException("contents");
             }
 
+            if (_htmlEncoder == null)
+            {
+                _htmlEncoder = context.ApplicationServices.GetRequiredService<IHtmlEncoder>();
```

Comment:
Consider using the `IServiceProvider.GetHtmlEncoder()` extension method instead. It will give you a default encoder instance if one hasn't already been registered in the service provider.